README.md

```diff
@@ -34,14 +43,143 @@ then `func` can discover it and expose it to a user in a similar way the static
+            "condition": {
+              "type": "equals",
+              "rhs": ".build-system",
+              "lhs": "'maven'"
+            }
+          }
+        }
+      ]
+    }
+  }
+  ```
+* The `create` sub-command shall accept four argument:
+  * --name: name of the Function project
+  * --language: language of the Function project
+  * --template: the template to create the Function project</br>
+  The sub-command shall use `stdio` in a specific way:</br></br>
+  On `stdin` there would be additional parameters provided as an JSON Object based on specification provided
+  by the `templates` sub-command for given language/template.</br>
+  Example of such a JSON:
+  ```json
+  {
+    "answers": [
+      {
+        "name": "artifact-id",
+        "value": "my-art"
+      },
+      {
+        "name": "group-id",
+        "value": "my.grp"
+      },
+      {
+        "name": "build-system",
+        "value": "gradle"
+      }
+    ]
+  }
+  ```
+  The sub-command is supposed to return project files as `tar` archive on `stdout`.
+
+The advantage of the `Contract B` is that it could be better for external tools,
+but implementation would be more difficult.
+
+## Containerization
+
+As mentioned before the `excutable` could be containerized.
+
+Advantages would be:
+* Easier distribution. Multi-arch images would make it easier to support multiple platforms.
+Without it users would have to download the righ binaries for themselves.
+* Better isolation. If the `executable` is not trusted it's safer to limit its access
+to filesystem, network and environment variables.</br>
+On the other hand this would limit `executable` no only in doing bad things but also good things.
+For instance template could ask `k8s` cluster about available event sources and ask users if they
+want to use some as a source for their function and generate code accordingly.
+
 ## Go Helper Library
 
-To ease development of binaries satisfying [executable contract](#executable-contract) we will offer `Go` library.
+To ease development of binaries satisfying `executable contract` we will offer `Go` library.
 
 The library is centered around
-[Repository](https://github.com/matejvasek/func-dynamic-templates/blob/main/lib/repository.go#L23) and
-[Template](https://github.com/matejvasek/func-dynamic-templates/blob/main/lib/repository.go#L16)
-interfaces.
+`Repository`and `Template` interfaces.
+
+<b>Contract A</b>
+```go
+// For contract A
+type Template interface {
+	Name() string
+	Runtime() string
+	// Write creates all project files including func.yaml
+	Write(ctx context.Context, projectName, destDir string) error
+}
+
+type Repository interface {
+	Runtimes(ctx context.Context) (runtimes []string, err error)
+	Templates(ctx context.Context, runtime string) (templates []string, err error)
+	Template(ctx context.Context, runtime, templateName string) (Template, error)
+}
+```
+
+<b>Contract B</b>
+```go
+// For contract B
+type Question interface {
+	// TODO
```

Comment:
Precisely specifying this won't be easy. We would need to think of various input types and possible dependencies between questions.

Reply:
This is one of my main concerns about Contract B. It seems like we'll have to take so many potential scenarios into account, and are bound to get some things wrong.